stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/actions/Close.java

```diff
@@ -42,6 +44,17 @@
 
     @Override
     public void execute(Context context) {
-        closeConnections(retrieveDatasource(context));
+        SQLDatasource datasource = retrieveDatasource(context);
+        // When an exception is thrown during database endpoint init (eg: driver not present) stop operation
+        // of the endpoint is automatically called. But at this point, datasource is null therefore to handle that
+        // situation following null check is needed.
+        if (datasource != null && !datasource.isGlobalDatasource()) {
+            try {
+                datasource.decrementClientCounter();
```

Comment:
Shall we rename the method `decrementClientCounter()` to something like `releaseConnectionPool()`. Decrementing is only a part of that operation.

Reply:
When a client decides to stop, datasource decrement the client counter and eventually when it reaches zero pool is shut down.
So isn't it more like, pool shutdown happens as a result of client counter reaching zero?

Reply:
As discussed offline, will rename the method to imply we are decrementing the client count and attempting to shut the pool down.